lib/phoenix/endpoint/cowboy_handler.ex

```diff
@@ -86,17 +86,35 @@ defmodule Phoenix.Endpoint.CowboyHandler do
   defp default_for(Phoenix.Transports.WebSocket), do: Phoenix.Endpoint.CowboyWebSocket
   defp default_for(_), do: nil
 
+  # Bake in retrieving the actual port from the adapter if possible, otherwise
+  # fallback to lower-level/less portable behavior.
+  defmacrop get_listener_port(ref) do
```

Comment:
To call attention to it; my hope is that it'll be removed once plug is certain to be exporting the info function. I can switch it over to a standard function if you'd like and, if you're okay with leaving it as-is, I've written in a little "remove me" note for the next commit on this PR.